numpy/linalg/linalg.py

```diff
@@ -1933,6 +1934,14 @@ def lstsq(a, b, rcond=-1):
     a, bstar = _fastCopyAndTranspose(t, a, bstar)
     a, bstar = _to_native_byte_order(a, bstar)
     s = zeros((min(m, n),), real_t)
+    # This line:
+    #  * is incorrect, according to the LAPACK documentation
+    #  * raises a ValueError if min(m,n) == 0
+    #  * should not be calculated here anyway, as LAPACK should calculate
+    #    `liwork` for us. But that only works if our version of lapack does
+    #    not have this bug:
+    #      http://icl.cs.utk.edu/lapack-forum/archives/lapack/msg00899.html
+    #    Lapack_lite does have that bug...
     nlvl = max( 0, int( math.log( float(min(m, n))/2. ) ) + 1 )
```

Comment:
According to LAPACK, this should be:

> SMLSIZ is returned by ILAENV and is equal to the maximum
> size of the subproblems at the bottom of the computation
> tree (usually about 25), and
>    NLVL = MAX( 0, INT( **LOG_2**( MIN( M,N )/(**SMLSIZ**+1) ) ) + 1 )

For whatever reason, we seem to have decided that `log(2) = SMLSIZ = 1`, which seems best described as "false". Either way, this is a can of worms, and not one that I think this PR should be opening.